lib/daru/index.rb

```diff
@@ -96,6 +96,27 @@ def [](*key)
         v
       end
     end
+    
+    def pos *args
+      # Causes Segmentation fault in array_wrapper
+      if args.first.is_a? Range
+        return args.first.to_a
```

Comment:
> NMatrix is supposed to store nils (:object dtype). 

So, in this case I think the bug is in Daru because I tried creating a vector with nil as dtype: :nmatrix and there was the same error.
```ruby
> Daru::Vector.new [2, nil], dtype: :nmatrix
```